keras/utils/audio_dataset.py

```diff
@@ -0,0 +1,401 @@
+    directory,
+    labels="inferred",
+    label_mode="int",
+    class_names=None,
+    batch_size=32,
+    sampling_rate=None,
+    output_sequence_length=None,
+    ragged=False,
+    shuffle=True,
+    seed=None,
+    validation_split=None,
+    subset=None,
+    follow_links=False,
+):
+    """Generates a `tf.data.Dataset` from audio files in a directory.
+
+    If your directory structure is:
+
+    ```
+    main_directory/
+    ...class_a/
+    ......a_audio_1.wav
+    ......a_audio_2.wav
+    ...class_b/
+    ......b_audio_1.wav
+    ......b_audio_2.wav
+    ```
+    
+    Then calling `audio_dataset_from_directory(main_directory, labels='inferred')`
+    will return a `tf.data.Dataset` that yields batches of audio files from
+    the subdirectories `class_a` and `class_b`, together with labels
+    0 and 1 (0 corresponding to `class_a` and 1 corresponding to `class_b`).
+    
+    Only `.wav` files are supported at this time.
+
+    Args:
+      directory: Directory where the data is located.
+          If `labels` is "inferred", it should contain
+          subdirectories, each containing audio files for a class.
+          Otherwise, the directory structure is ignored.
+      labels: Either "inferred"
+          (labels are generated from the directory structure),
+          None (no labels),
+          or a list/tuple of integer labels of the same size as the number of
+          audio files found in the directory. Labels should be sorted according
+          to the alphanumeric order of the audio file paths
+          (obtained via `os.walk(directory)` in Python).
+      label_mode: String describing the encoding of `labels`. Options are:
+          - 'int': means that the labels are encoded as integers
+              (e.g. for `sparse_categorical_crossentropy` loss).
+          - 'categorical' means that the labels are
+              encoded as a categorical vector
+              (e.g. for `categorical_crossentropy` loss).
+          - 'binary' means that the labels (there can be only 2)
+              are encoded as `float32` scalars with values 0 or 1
+              (e.g. for `binary_crossentropy`).
+          - None (no labels).
+      class_names: Only valid if "labels" is "inferred". This is the explicit
+          list of class names (must match names of subdirectories). Used
+          to control the order of the classes
+          (otherwise alphanumerical order is used).
+      batch_size: Size of the batches of data. Default: 32.
+        If `None`, the data will not be batched
+        (the dataset will yield individual samples).
+      sampling_rate: Number of samples taken each second.
+      output_sequence_length: Maximum length of a audio sequence. audio files longer than this will 
+      be truncated to `output_sequence_length`
+      ragged: Whether to return a ragged dataset if the sequence length is not the same. Default: False.
+          If set to False, `output_sequence_length` should have a value.
```

Comment:
Upon further consideration, I'm not sure about this -- having `ragged=False, output_sequence_length=None` should be valid and should result in a sequence length padded to the longest sequence in the current batch. This is what happens in `TextVectorization` in `int` mode. WDYT?

Reply:
Yes I think it makes sense to pad to the longest sequence in the current batch, I will modify this part to use `padded_batch` instead of `batch` if `output_sequence_length` is `None`